src/main/java/com/epam/dataservice/AccidentBatchProcessor.java

```diff
@@ -0,0 +1,97 @@
+package com.epam.dataservice;
+
+import com.epam.data.RoadAccident;
+
+import java.util.List;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Created by bill on 16-5-1.
+ */
+public class AccidentBatchProcessor  implements Runnable {
+    private static AtomicInteger serialNo = new AtomicInteger();
+// For record Enrichment
+    private static PoliceForceService policeForceService = new PoliceForceService();
+    private static ExecutorService enrichExecutor = Executors.newFixedThreadPool(HomeWork3.MAX_ENRICH_THREADS);
+    private CompletionService<RoadAccident> enrichCompletionService = new ExecutorCompletionService<RoadAccident>(enrichExecutor);
+
+    private String taskName = getClass().getSimpleName() + serialNo.incrementAndGet();
+    private BlockingQueue<List<RoadAccident>> readQueue;
+    private List<BlockingQueue<RoadAccident>> writeQueues;
+
+    public  AccidentBatchProcessor(BlockingQueue<List<RoadAccident>> readQueue, List<BlockingQueue<RoadAccident>> writeQueues){
```

Comment:
Let's imagine you write such code on production.
It's a public constructor, so someone else potentially might be using it (maybe your teammate, or someone who will come on project after your) 

For such user the constructor will look quite confusing. read queue is queue of lists, which makes sense, but writequeue is list of queues :confused: 

Most probably I will have to read a code to understand such transformation. 

In future, try to be consistent in your APIs. it would make more sense to me to accept and produce similar type of objects.